File: 7KYU/Composing squared strings/solution.clj
(ns compsqstrings.core
  (:require [clojure.string :as str]))

(defn compose [s1 s2]
  (let [s1 (str/split-lines s1)
        s2 (str/split-lines s2)
        sq (count s1)]
    (->> (map flatten
              (for [r (range 0 sq)]
                [(take (inc r) (nth s1 r)) (take (- sq r) (nth s2 (- sq (inc r))))]))
         (map (partial apply str))
         (str/join "\n"))))

File: 7KYU/Composing squared strings/solution.go
package kata

import "strings"

func Compose(s1, s2 string) string {
	s1lines := strings.Split(s1, "\n")
	s2lines := strings.Split(s2, "\n")

	n := len(s1lines)
	res := []string{}
	for i := 0; i < n; i++ {
		s := string(s1lines[i][:i+1]) + s2lines[n-1-i][0:n-i]
		res = append(res, s)
	}

	return strings.Join(res, "\n")
}
